certutils/certutils.go

```diff
@@ -67,13 +70,18 @@ type TLSServerConfig struct {
 }
 
 // NewTLSServer sets up a Pantheon(TM) type of tls server that Requires and Verifies peer cert
-func NewTLSServer(config TLSServerConfig) *http.Server {
+func NewTLSServer(config TLSServerConfig, requireClientCert *bool) *http.Server {
+	if requireClientCert == nil {
+		requireClientCert := RequireClientCertDefault
+	}
 	// Setup our TLS config
 	tlsConfig := NewTLSConfig(config.TLSConfigLevel)
 
 	// By default this server will require client MTLS certs and verify cert validity against the config.CertPool CA bundle
-	tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
-	tlsConfig.ClientCAs = config.CertPool
+	if true == *requireClientCert {
```

Comment:
If we were going to add the ability to the go-certauth to set up TLS *without* certauth, what I would recommend instead of a bool is:

- Have this method, `NewTLSServer`, keep its existing paramers
- Add a new method, maybe `NewTLSServerWithConfig`, that takes a `tlsConfig` as a parameter. Call `NewTLSServerWithConfig(tlsConfig)` here-ish, after adding the ClientAuth and ClientCAs.
- Clients that don't want certauth would then just call `NewTLSServerWithConfig` instead of `NewTLSServer`, and pass in the `tlsConfig` they fetched from the `NewTLSConfig` method.

Reply:
Good idea. Will refactor first thing in the morning, if not before that.